libraries/app/database_api.cpp

```diff
@@ -1253,6 +1254,58 @@ market_volume database_api_impl::get_24_volume( const string& base, const string
     return result;
 }
 
+vector<market_volume> database_api::get_top_markets(uint start, uint limit)const
+{
+   return my->get_top_markets(start, limit);
+}
+
+vector<market_volume> database_api_impl::get_top_markets(uint start, uint limit)const
+{
+   FC_ASSERT( limit <= 100 );
+
+   const auto& volume_idx = _db.get_index_type<graphene::market_history::market_ticker_index>().indices().get<by_volume>();
+   auto itr = volume_idx.rbegin();
+   vector<market_volume> result;
+
+   fc::uint128 base_volume;
+   fc::uint128 quote_volume;
+
+   uint c = 0;
+   const fc::time_point_sec now = fc::time_point::now();
+
+   while( itr != volume_idx.rend())
+   {
+      if(c < start) {
+         ++c;
+         ++itr;
+         continue;
```

Comment:
This is inefficient. Perhaps we just remove the `start` parameter and enlarge maximum of `limit` since the data can be highly volatile which will make it hard to do pagination.